x-pack/plugins/security_solution/public/timelines/components/side_panel/index.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { EuiFlyout } from '@elastic/eui';
+import styled from 'styled-components';
+import { timelineActions, timelineSelectors } from '../../store/timeline';
+import { timelineDefaults } from '../../store/timeline/defaults';
+import { BrowserFields, DocValueFields } from '../../../common/containers/source';
+import { TimelineTabs } from '../../../../common/types/timeline';
+import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { EventDetailsPanel } from './event_details';
+import { HostDetailsPanel } from './host_details';
+import { NetworkDetailsPanel } from './network_details';
+
+const StyledEuiFlyout = styled(EuiFlyout)`
+  z-index: ${({ theme }) => theme.eui.euiZLevel7};
+`;
+
+interface DetailsPanelProps {
+  browserFields: BrowserFields;
+  docValueFields: DocValueFields[];
+  handleOnPanelClosed?: () => void;
+  isFlyoutView?: boolean;
+  tabType?: TimelineTabs;
+  timelineId: string;
+}
+
+/**
+ * This panel is used in both the main timeline as well as the flyouts on the host, detection, cases, and network pages.
+ * To prevent duplication the `isFlyoutView` prop is passed to determine the layout that should be used
+ * `tabType` defaults to query and `handleOnPanelClosed` defaults to unsetting the default query tab which is used for the flyout panel
+ */
+export const DetailsPanel = React.memo(
+  ({
+    browserFields,
+    docValueFields,
+    handleOnPanelClosed,
+    isFlyoutView,
+    tabType,
+    timelineId,
+  }: DetailsPanelProps) => {
+    const dispatch = useDispatch();
+    const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+    const expandedDetail = useDeepEqualSelector((state) => {
+      return (getTimeline(state, timelineId) ?? timelineDefaults).expandedDetail;
+    });
+
+    // To be used primarily in the flyout scenario where we don't want to maintain the tabType
+    const defaultOnPanelClose = useCallback(() => {
+      dispatch(timelineActions.toggleDetailPanel({ timelineId }));
+    }, [dispatch, timelineId]);
+
+    const activeTab = useMemo(() => tabType ?? TimelineTabs.query, [tabType]);
```

Comment:
we do not need useMemo here, it is a string

Reply:
Removed :)